src/docs/development/platform-integration/web-images.md

```diff
@@ -0,0 +1,161 @@
+image. Therefore, images to be rendered using WebGL must only come from servers
+that have a CORS policy configured to work with the domain that serves your
+application.
+
+## Flutter renderers on the web
+
+Flutter offers a choice of two renderers on the web:
+
+- **HTML**: this renderer uses a combination of HTML, CSS, Canvas 2D, and SVG
+  to render UI. It uses the `<img>` element to render images.
+- **CanvasKit**: this renderer uses WebGL to render UI, and therefore requires
+  access to the pixels of the image.
+
+Because the HTML renderer uses the `<img>` element it can display images from
+arbitrary sources. However, this places the following limitations on what you
+can do with them:
+
+- Limited support for [Image.toByteData][7].
+- No support for [OffsetLayer.toImage][8] and [Scene.toImage][10].
+- No access to frame data in animated images ([Codec.getNextFrame][9],
+  `frameCount` is always 1, `repetitionCount` is always 0).
+- No support for `ImageShader`.
+- Limited support for shader effects that can be applied to images.
+- No control over image memory (`Image.dispose` has no effect). The memory is
+  managed by the browser behind-the-scenes.
+
+The CanvasKit renderer implements Flutter's image API fully. However, it
+requires access to image pixels to do so, and is therefore subject to the CORS
+policy.
+
+# Solutions
+
+## In-memory, asset, and same-origin network images
+
+If the app has the bytes of the encoded image in memory, provided as an
+[asset][12], or stored on the same server that serves the application (a.k.a.
+same-origin), no extra effort is necessary. The image can be displayed using
+the [Image.memory][13], [Image.asset][14], and [Image.network][15] in both HTML
+and CanvasKit modes.
+
+## Cross-origin images
+
+The HTML renderer can load cross-origin images without extra configuration.
+
+CanvasKit requires that the app get the bytes of the encoded image. There are
+several ways to do this, discussed below.
+
+### Host your images in a CORS-enabled CDN.
+
+Typically content delivery networks (CDN) can be configured to customize what
+domains are allowed to access your content. For example, Firebase site hosting
+allows [specifying a custom][16] `Access-Control-Allow-Origin` header in the
+`firebase.json` file.
+
+### Lack control over the image server? Use a CORS proxy.
+
+If the image server cannot be configured to allow CORS requests from your
+application, you may still be able to load images by proxying the requests
+through another server. This will require that the intermediate server has
+sufficient access to load the images.
+
+This method can be used in situations when the original image server serves
+images publicly, but is not configured with the correct CORS headers.
+
+Examples:
+
+- Using [CloudFlare Workers][18].
+- Using [Firebase Functions][19].
+
+### Use <img> in a platform view.
+
+Flutter supports embedding HTML inside the app using [HtmlElementView][17]. Use
```

Comment:
Filed an issue to add sample code for this: https://github.com/flutter/website/issues/5063